pkg/probe/probes.go

```diff
@@ -77,174 +77,165 @@ func currentStateAsGJson() (gjson.Result, error) {
 	return gjson.ParseBytes(currentState), nil
 }
 
-func ping(target string, timeout time.Duration) (string, error) {
-	output := ""
-	return output, wait.PollImmediate(time.Second, timeout, func() (bool, error) {
-		cmd := exec.Command("ping", "-c", "1", target)
-		var outputBuffer bytes.Buffer
-		cmd.Stdout = &outputBuffer
-		cmd.Stderr = &outputBuffer
-		err := cmd.Run()
-		output = fmt.Sprintf("cmd output: '%s'", outputBuffer.String())
-		if err != nil {
-			return false, nil
-		}
-		return true, nil
-	})
+func ping(target string) (string, error) {
+	cmd := exec.Command("ping", "-c", "1", target)
+	var outputBuffer bytes.Buffer
+	cmd.Stdout = &outputBuffer
+	cmd.Stderr = &outputBuffer
+	err := cmd.Run()
+	output := fmt.Sprintf("cmd output: '%s'", outputBuffer.String())
+	if err != nil {
+		return "", errors.Wrapf(err, "failed running ping probe: %s", output)
+	}
+	return output, nil
 }
 
-// This probes use its own client to bypass cache that
-// why we wrap it to ignore the one it's passed
-func checkAPIServerConnectivity(timeout time.Duration) error {
-	return wait.PollImmediate(time.Second, timeout, func() (bool, error) {
-		// Create new custom client to bypass cache [1]
-		// [1] https://github.com/operator-framework/operator-sdk/blob/master/doc/user/client.md#non-default-client
-		currentConfig, err := config.GetConfig()
-		if err != nil {
-			return false, errors.Wrap(err, "getting config")
-		}
-		// Since we are going to retrieve Nodes default schema is good
-		// enough, also align timeout with poll
-		currentConfig.Timeout = timeout
-		cli, err := client.New(currentConfig, client.Options{})
-		if err != nil {
-			log.Error(err, "failed to creating new custom client")
-			return false, nil
-		}
-		err = cli.Get(context.TODO(), types.NamespacedName{Name: metav1.NamespaceDefault}, &corev1.Namespace{})
-		if err != nil {
-			log.Error(err, "failed reaching the apiserver")
-			return false, nil
-		}
-		return true, nil
-	})
+// This probes use its own client to bypass cache
+func checkAPIServerConnectivity() error {
+	// Create new custom client to bypass cache [1]
+	// [1] https://github.com/operator-framework/operator-sdk/blob/master/doc/user/client.md#non-default-client
+	currentConfig, err := config.GetConfig()
+	if err != nil {
+		return errors.Wrap(err, "getting config")
+	}
+
+	// Since we are going to retrieve Nodes default schema is good
+	// enough, also align timeout with poll timeout
+	currentConfig.Timeout = apiServerProbeTimeout
+	cli, err := client.New(currentConfig, client.Options{})
```

Comment:
Do we really need this one ? I supose that we have similar PollImmediate behind the scenes

Reply:
Fixed by setting the client timeout to 0 (to disable it)